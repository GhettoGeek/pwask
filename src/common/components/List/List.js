import React from 'react'
import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next'
import {
  List, ListItem, ListItemIcon, ListItemText,
} from '@material-ui/core'
import Link from '../Link'
import Icon from '../Icon'

function EnhancedList({ items }) {
  const { t } = useTranslation()

  return (
    <List>
      {
        items.map((item) => (
          <ListItem button key={t(item.label)}>
            {item.icon && (
              <ListItemIcon>
                <Icon name={item.icon} />
              </ListItemIcon>
            )}
            <Link to={item.link}>
              <ListItemText primary={t(item.label)} />
            </Link>
          </ListItem>
        ))
      }
    </List>
  )
}

EnhancedList.propTypes = {
  items: PropTypes.arrayOf(PropTypes.object).isRequired,
}

export default EnhancedList
