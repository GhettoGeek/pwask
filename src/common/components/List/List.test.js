import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { shallow } from 'enzyme'
import List from './List'

const items = [
  {
    label: 'item1',
    link: '/item1',
  },
  {
    label: 'item2',
    link: '/item2',
  },
]

describe('List', () => {
  it('renders correctly the component', () => {
    const wrapper = shallow(
      <MemoryRouter initialEntries={['/']} initialIndex={0}>
        <List items={items} />
      </MemoryRouter>
    )

    expect(wrapper.html()).toMatchSnapshot()
  })
})
