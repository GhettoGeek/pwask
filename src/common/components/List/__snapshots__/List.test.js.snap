// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly the component 1`] = `"<ul class=\\"MuiList-root MuiList-padding\\"><div class=\\"MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary\\" href=\\"/item1\\"><div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1\\">item1</span></div></a></div><div class=\\"MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary\\" href=\\"/item2\\"><div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1\\">item2</span></div></a></div></ul>"`;
